### PR TITLE
[Atuin] Module 

### PR DESCRIPTION
**Pull request by Tearran**
Interesting title. Playing with making a module for  basic setup of atuin. 
`bash "./tools/modules/software/module_atuin" help`
>note missing .sh 
assemble looks for *.sh before writing to the lib
can leverage this for testing so its not included
```
Commands: help install remove status
```
```
Available commands:
  install       - Install Atuin 
```
https://docs.atuin.sh/guide/installation/

running installs  to $HOME

Not sure how to make this global. maybe module `update_skel` may be a hint?

### CODE REVIEW

**Comment by EvilOlaf** [comment]
Some thoughts about that from the chat:
- beohoff mentions the latency for bash users is better when using bash-preexec instead of ble.sh. I did not research which one is installed by default by atuin.
- I've setup our own server for sync: https://services.armbian.de/atuin . This is subject of change, like using a different unused domain for this or whatever.
- - using environmental variable for sync server only works for `register` command but once the default config is written (which happens AFTER issuing `register`) it seems superiour. Also environment variable isn't written back into this config so its either about us or the user to adjust sync address... if they prefer to do so.

**Comment by Tearran** [comment]
>beohoff mentions the latency for bash users is better when using bash-preexec instead of ble.sh. I did not research which one is installed by default by atuin.

Seem like it is using  `preexec`, The installer does use `~/.bash-preexec.sh` and no referance to ble.sh

**Comment by Tearran** [comment]
>Therefore when implementing this via configng the config file with adjustments needs to be written the installation (line 32: sync_address = "https://services.armbian.de/atuin")

after install the config looks all commented. so we can overwrite the default and use a custom one.

to restore the default
`atuin default-config > ~/.config/atuin/config.toml`

**Comment by EvilOlaf** [comment]
> after install the config looks all commented. so we can overwrite the default and use a custom one.

Yes, most seems commented by default. For me the following ones were uncommented by default:
`enter_accept = true` and `records = true`

**Comment by Tearran** [comment]
>Yes, most seems commented by default. For me the following ones were uncommented by default:
enter_accept = true and records = true


This seems to cover the default variables for the sync server
```
sync_address = "https://services.armbian.de/atuin" 
enter_accept = true

[sync]
records = true
```
Included, IMO to be relative in continued development 
paths and daemon settings

This still only able to work in "$HOME" "~/" environment.

**Comment by Tearran** [comment]
latest update. 
works with bash 
added .sh to include
```
./tools/config-assemble.sh" -p
```
```
./bin/armbian-config --api module_atuin help
```
```

Usage: module_atuin <command>
Commands: help install remove reset

Available commands:
  install       - setup atuin in /home/tearran
  reset         -  reset config to atuin default
  remove        -  Uninstall atuin
```
```
 ./configng/bin/armbian-config --api see_menu module_atuin
```
image
install 

restart shell

```
atuin help
atuin 18.4.0-beta.3
Ellie Huxtable <ellie@atuin.sh>
Magical shell history

Usage:
  atuin <COMMAND>

Commands:
  history          Manipulate shell history
  import           Import shell history from file
  stats            Calculate statistics for your history
  search           Interactive history search
  sync             Sync with the configured server
  login            Login to the configured server
  logout           Log out
  register         Register with the configured server
  key              Print the encryption key for transfer to another machine
  status           Display the sync status
  account          Manage your sync account
  kv               Get or set small key-value pairs
  store            Manage the atuin data store
  dotfiles         Manage your dotfiles with Atuin
  init             Print Atuin's shell init script
  info             Information about dotfiles locations and ENV vars
  doctor           Run the doctor to check for common issues
  daemon           *Experimental* Start the background daemon
  default-config   Print the default atuin configuration (config.toml)
  server           Start an atuin server
  uuid             Generate a UUID
  contributors
  gen-completions  Generate shell completions
  help             Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help
  -V, --version  Print version
```
`AROW UP`
image

https://docs.atuin.sh

**Comment by Tearran** [comment]
Issues:
- Not an improvement to the installer @atuinsh provides. 
- sudo make checking and installing overcomplicate. 
-  only supports bash, config for other need to be added

**Comment by igorpecovnik** [comment]
> only supports bash, config for other need to be added

ZSH, as we provide it, would be nice to work OOB.

> Not an improvement to the installer
> sudo make checking and installing overcomplicate.

What if we add this functionality system-wide, like our ZSH implementation. Superuser is even changing default shell for all. If Admin of OS do this, user can move back to BASH at any moment. We can do one step less here, to do everything for the users of the system, except enabling it. And give instructions how to enable this feature with a simple (user) step.

https://github.com/armbian/configng/blob/main/tools/modules/system/update_skel.sh

File: tools/modules/software/module_atuin.sh [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	["module_atuin,author"]="@atuinsh"
+	["module_atuin,maintainer"]="@armbian @Tearran"
+	["module_atuin,testers"]="@Tearran"
+	["module_atuin,feature"]="module_atuin"
+	["module_atuin,example"]="help install remove reset"
+	["module_atuin,desc"]="Module to install logout and uninstall Atuin."
+	["module_atuin,remote"]="https://services.armbian.de/atuin"
+	["module_atuin,port"]=""
+	["module_atuin,status"]="review"
+)
```
[the remark this answers is not in the document]
Comment (Tearran) [reply tools/modules/software/module_atuin.sh:14]: library, module_options is declared before sourcing

**Comment by EvilOlaf** [comment]
version should be bumped to v18.5.0

**Comment by Tearran** [comment]
Closing Stalled but not forgotten. Considering configuration approach

**Comment by EvilOlaf** [comment]
Server version was bumped to 18.6.0

**Comment by igorpecovnik** [comment]
_"This is works for me including the timezone for check logs across along on the clients. Validates in Raspbery pi 4 with **Armbian** 25.2.2 Bookworm"_
https://forum.atuin.sh/t/running-atuin-on-a-raspberry-pi-with-docker-compose/914

**Pull request closed without merging**